odoo/addons/base/models/ir_sequence.py

```diff
@@ -53,8 +53,14 @@ def _select_nextval(cr, seq_name):
 
 def _update_nogap(self, number_increment):
     number_next = self.number_next
-    self._cr.execute("SELECT number_next FROM %s WHERE id=%%s FOR UPDATE NOWAIT" % self._table, [self.id])
-    self._cr.execute("UPDATE %s SET number_next=number_next+%%s WHERE id=%%s " % self._table, (number_increment, self.id))
+    try:
+        self._cr.execute("SELECT number_next FROM %s WHERE id=%%s FOR UPDATE NOWAIT" % self._table, [self.id])
+        self._cr.execute("UPDATE %s SET number_next=number_next+%%s WHERE id=%%s " % self._table, (number_increment, self.id))
+    except OperationalError as e:
+        if e.pgcode == errorcodes.LOCK_NOT_AVAILABLE:
```

Comment:
Hello, this code break the automatic retry when a concurrency exception is raised by psycopg2.

Here => https://github.com/odoo/odoo/blob/e2b5a359c1d5eccbe725c1c3169b4130d7bca49b/odoo/service/model.py#L103

Is it normal?

Reply:
Ouch, probably not :facepalm:.  Thanks for the notice.

@rim-odoo Do we consider it's acceptable to ask the user to retry manually (instead of the server default retry) in this case ?
If not, I should revert this current change and we could think of giving "user-understandable translated" psycopg messages at the `service/model.py` level, but it would be less precise because not specific to `no_gap` sequences.

Reply:
Hello, why not catch the exception when done from a user interaction (a button, an action, etc ...?)

Here I think this should be reverted as this code is called from multiple places that rely on the automatic retry mechanism as the no gap sequences are really affected by the concurrency issues.

Reply:
I'll revert it tomorrow and try to find a solution more targeted (while keeping the automatic retry as much as possible) ;).

Reply:
Arf shit, (I was off last week), sorry for the disturbance @tbe-odoo ; no one noticed that it would break the re-try :'(

Reply:
No problem, not an easy one to spot :laughing: